# Laboratorium 2 #
#### Zofia Jankowska ####

In [43]:
import numpy as np
import scipy
import matplotlib
import matplotlib.pyplot as plt
import pandas as pd

In [44]:
from sqlalchemy import create_engine
import psycopg2 as pg
import pandas as pd

db_string = "postgresql://wbauer_adb:adb2020@pgsql-196447.vipserv.org:5432/wbauer_adb"
connection = pg.connect(host='pgsql-196447.vipserv.org', port=5432, dbname='wbauer_adb', user='wbauer_adb', password='adb2020')

db = create_engine(db_string)

connection_sqlalchemy = db.connect()

**Zadanie 1.** <br>
Ile kategorii filmów mamy w wypożyczalni?

In [45]:
df_1 = pd.read_sql('select * from category',con=connection_sqlalchemy)
print(df_1)


    category_id         name         last_update
0             1       Action 2006-02-15 09:46:27
1             2    Animation 2006-02-15 09:46:27
2             3     Children 2006-02-15 09:46:27
3             4     Classics 2006-02-15 09:46:27
4             5       Comedy 2006-02-15 09:46:27
5             6  Documentary 2006-02-15 09:46:27
6             7        Drama 2006-02-15 09:46:27
7             8       Family 2006-02-15 09:46:27
8             9      Foreign 2006-02-15 09:46:27
9            10        Games 2006-02-15 09:46:27
10           11       Horror 2006-02-15 09:46:27
11           12        Music 2006-02-15 09:46:27
12           13          New 2006-02-15 09:46:27
13           14       Sci-Fi 2006-02-15 09:46:27
14           15       Sports 2006-02-15 09:46:27
15           16       Travel 2006-02-15 09:46:27


In [46]:
df_1 = pd.read_sql('select count(category_id) from category',con=connection_sqlalchemy)
print(df_1)

   count
0     16


Jest 16 kategorii filmów.

**Zadanie 2.** <br>
Wyświetl listę kategorii w kolejności alfabetycznej. <br>

In [47]:
df_2 = pd.read_sql('select name from category order by name',con=connection_sqlalchemy)
print(df_2)

           name
0        Action
1     Animation
2      Children
3      Classics
4        Comedy
5   Documentary
6         Drama
7        Family
8       Foreign
9         Games
10       Horror
11        Music
12          New
13       Sci-Fi
14       Sports
15       Travel


**Zadanie 3.** <br>
Znajdź najstarszy i najmłodszy film do wypożyczenia.

In [48]:
df_3 = pd.read_sql('select title, release_year from film order by release_year',con=connection_sqlalchemy)
print(df_3)

                 title  release_year
0      Chamber Italian          2006
1     Grosse Wonderful          2006
2      Airport Pollock          2006
3    Bright Encounters          2006
4     Academy Dinosaur          2006
..                 ...           ...
995     Young Language          2006
996         Youth Kick          2006
997       Zhivago Core          2006
998  Zoolander Fiction          2006
999          Zorro Ark          2006

[1000 rows x 2 columns]


Wszystkie filmy zostały wydane w 2006, więc nie ma najstarszego lub najmłodszego filmu do wypożyczenia.

**Zadanie 4.** <br>
Ile wypożyczeń odbyło się między 2005-07-01 a 2005-08-01? 

In [59]:
df_4 = pd.read_sql('select rental_date from rental where rental_date between str_to_date("2005-07-01 00:00:00", "%yyyy-%mm-%dd") and 2005-08-01',con=connection_sqlalchemy)
print(df_4)


ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "2005"
LINE 1: ..._date from rental where rental_date between dates 2005-07-01...
                                                             ^

[SQL: select rental_date from rental where rental_date between dates 2005-07-01 and 2005-08-01]
(Background on this error at: https://sqlalche.me/e/14/f405)

**Zadanie 5.** <br>
Ile wypożyczeń odbyło się między 2010-01-01 a 2011-02-01? <br>

**Zadanie 6.** <br>
Znajdź największą płatność wypożyczenia. <br>

In [53]:
df_4 = pd.read_sql('select max(amount) from payment',con=connection_sqlalchemy)
print(df_4)

     max
0  11.99


Maksymalna płatnośc wyniosła 11.99

**Zadanie 7.** <br>
Znajdź wszystkich klientów z Polski, Nigerii lub Bangladeszu. <br>

**Zadanie 8.** <br>
Gdzie mieszkają członkowie personelu? <br>

In [57]:
df_4 = pd.read_sql('select * from address',con=connection_sqlalchemy)
print(df_4)

     address_id                   address address2      district  city_id  \
0             1         47 MySakila Drive     None       Alberta      300   
1             2        28 MySQL Boulevard     None           QLD      576   
2             3         23 Workhaven Lane     None       Alberta      300   
3             4      1411 Lillydale Drive     None           QLD      576   
4             5            1913 Hanoi Way               Nagasaki      463   
..          ...                       ...      ...           ...      ...   
598         601       844 Bucuresti Place               Liaoning      242   
599         602  1101 Bucuresti Boulevard            West Greece      401   
600         603    1103 Quilmes Boulevard                  Piura      503   
601         604       1331 Usak Boulevard                   Vaud      296   
602         605      1325 Fukuyama Street           Heilongjiang      537   

    postal_code         phone         last_update  
0                      

**Zadanie 9.** <br>
Ilu pracowników mieszka w Argentynie lub Hiszpanii? <br>

**Zadanie 10.** <br>
Jakie kategorie filmów zostały wypożyczone przez klientów? <br>

**Zadanie 11.** <br>
Znajdź wszystkie kategorie filmów wypożyczonych w Ameryce. <br>

**Zadanie 12.** <br>
Znajdź wszystkie tytuły filmów, w których grał: Olympia Pfeiffer lub Julia  Zellweger lub Ellen Presley <br>